src/main/java/it/pagopa/qi/jsmmanagement/JsmIngestion.java

```diff
@@ -0,0 +1,103 @@
+     * @param jiraRestClientConfig - the jira rest client configuration class to be used
+     * @param jiraRestClient       - the jira rest client to be used
+     */
+    JsmIngestion(Logger logger, ObjectMapper objectMapper, JiraRestClientConfig jiraRestClientConfig, JiraRestClient jiraRestClient) {
+        this.logger = logger;
+        this.objectMapper = objectMapper;
+        this.jiraRestClientConfig = jiraRestClientConfig;
+        this.jiraRestClient = jiraRestClient;
+    }
+
+    @FunctionName("EventHubQiJsmProcessor")
+    public void processJsmAlert(
+            @EventHubTrigger(
+                    name = "QiJsmEvent",
+                    eventHubName = "", // blank because the value is included in the connection string
+                    connection = "EVENTHUB_CONN_STRING",
+                    cardinality = Cardinality.ONE)
+            String alertBody,
+            final ExecutionContext context) {
+        try {
+            logger.info("Received new alert webhook trigger request with invocation id: [{}]", context.getInvocationId());
+            Alert alert = objectMapper.readValue(alertBody, Alert.class);
+            logger.info("JSM function called at {} with alert {}", LocalDateTime.now(), alert);
+            IssueRestClient issueClient = jiraRestClient.getIssueClient();
+            String projectKey = jiraRestClientConfig.getPpiProjectId();
+            long issueTypeId = jiraRestClientConfig.getPpiIssueTypeId();
+            String environment = jiraRestClientConfig.getEnvironment();
+            AlertDetails alertDetails = alert.getDetails();
+            String ticketSummary = "A new KPI %s alert was triggered for %s".formatted(alertDetails.getCode(), alertDetails.getOwner());
+            String ticketDescription =
+                    """
+                            Alert details:
+                            KPI code: [%s]
+                            Owner: [%s]
+                            Threshold: [%s]
+                            Value: [%s]
+                            Alert triggering date: [%s]
+                            """.formatted(alertDetails.getCode(), alertDetails.getOwner(), alertDetails.getThreshold(), alertDetails.getValue(), alertDetails.getTriggerDate());
+            IssueInput newIssue = new IssueInputBuilder(
+                    projectKey, issueTypeId, "[QI - pagoPA] - [%s] %s".formatted(environment, ticketSummary))
+                    .setFieldValue("description", ticketDescription)
+                    .build();
+
+            String ticketId = issueClient.createIssue(newIssue).claim().getKey();
+            logger.info("Ticket created with ID = {}", ticketId);
+        } catch (JsonProcessingException e) {
+            logger.error("Error reading value", e);
+            throw new AlertParsingException(alertBody);
+        } catch (RestClientException e) {
+            logger.error("Error opening ticket to JSM", e);
```

Comment:
This block is never thrown by the methods we call (if we remove it no error is raised)

Reply:
`RestClientException` is a `RuntimeException` (the reason behind the fact that catch clause can be removed without compilation error) thrown by com.atlassian.jira.rest.client.api.IssueRestClient.createIssue method.
The purpose here is to catch that exception in order to properly log error raised opening issue on Jira